phpBB/includes/ucp/ucp_register.php

```diff
@@ -195,22 +195,32 @@ function main($id, $mode)
 
 					'COOKIE_NAME'		=> $config['cookie_name'],
 					'COOKIE_PATH'		=> $config['cookie_path'],
-					)
 				);
 			}
-			unset($lang_row);
+
+			$template->assign_vars($template_vars);
+
+			$tpl_name = 'ucp_agreement';
 
 			/**
 			* Allows to modify the agreements.
 			*
 			* To assign data to the template, use $template->assign_vars()
 			*
 			* @event core.ucp_register_agreement
+			* @var	string	tpl_name			Template file
+			* @var	array	template_data		Array with data assigned to the template, read only
+			* @var	array	lang_row			Array with available languages, read only
+			* @var	array	s_hidden_fields		Array with hidden form elements, read only
 			* @since 3.1.6-RC1
+			* @changed 3.2.2-RC1 Added tpl_name, template_data, lang_row, s_hidden_fields
 			*/
-			$phpbb_dispatcher->dispatch('core.ucp_register_agreement');
+			$vars = array('tpl_name', 'template_data', 'lang_row', 's_hidden_fields');
```

Comment:
`template_data` should be changed to `template_vars`. (Don't forget to update it in the comment as well)